test/e2e/l7networkpolicy_test.go

```diff
@@ -177,20 +178,24 @@ func probeL7NetworkPolicyHTTP(t *testing.T, data *TestData, serverPodName, clien
 	}
 }
 
-func probeL7NetworkPolicyTLS(t *testing.T, data *TestData, clientPodName string, serverName string, canAccess bool) {
-	url := fmt.Sprintf("https://%s", serverName)
-	assert.Eventually(t, func() bool {
-		cmd := []string{"wget", "-O", "-", url, "-T", "5"}
-		stdout, stderr, err := data.RunCommandFromPod(data.testNamespace, clientPodName, agnhostContainerName, cmd)
-		if canAccess && err != nil {
-			t.Logf("Failed to access %s: %v\nStdout: %s\nStderr: %s\n", url, err, stdout, stderr)
-			return false
-		} else if !canAccess && err == nil {
-			t.Logf("Expected not to access the server, but the request succeeded.\nStdout: %s\nStderr: %s\n", stdout, stderr)
-			return false
-		}
-		return true
-	}, 10*time.Second, time.Second)
+func probeL7NetworkPolicyTLS(t *testing.T, data *TestData, clientPodName string, serverNames []string, canAccess bool) {
+	for _, serverName := range serverNames {
+		url := fmt.Sprintf("https://%s", serverName)
+		assert.Eventually(t, func() bool {
+			// The built-in certificate of the test HTTPS server Pod does not include the test server names.
+			// Therefore, the test client Pod should not verify the test HTTPS server's certificate.
+			cmd := []string{"wget", "-O", "-", url, "-T", "1", "--no-check-certificate"}
```

Comment:
You can specify SNI via `--header="Host: <SERVER NAME>"` then you don't need to override `/etc/hosts`

Reply:
I have tried this, but it didn't work. I changed to use `curl`.

Reply:
Like this: `curl -k  --resolve www.alfa.test.l7.tls:443:10.10.0.41 https://www.alfa.test.l7.tls`